oauthlib/oauth2/rfc6749/endpoints/resource.py

```diff
@@ -72,7 +74,19 @@ def verify_request(self, uri, http_method='GET', body=None, headers=None,
                                              self.default_token_type_handler)
         log.debug('Dispatching token_type %s request to %r.',
                   request.token_type, token_type_handler)
-        return token_type_handler.validate_request(request), request
+        is_valid = token_type_handler.validate_request(request)
+
+        # the validate_request() call is a chance for the client library to finalize the scopes on the request
```

Comment:
Reviewing the flow here it looks like my comment 

> the validate_request() call is a chance for the client library to finalize the scopes on the request 

there is wrong.  `request.scopes` ought to be the scopes _required to access_ the Resource and can be defined by the client library prior to calling our `verify_request()`.  It's a bit confusing - if anything I'd rename the `scopes` argument of `verify_request()` to be `required_scopes`.  That would only slightly clarify it though - the `request.scopes` property is overloaded to mean both "scopes listed in the incoming HTTP request" in other request flows like Authorization and Token requests and "scopes required to access the requested resource" as in this protected resource request flow.